### DISCUSSION TITLE
PSRule.Rules.Azure/creating-your-pipeline/

### DISCUSSION THREAD

**Discussion by giscus** [Azure/PSRule.Rules.Azure#1895] [Documentation]
# PSRule.Rules.Azure/creating-your-pipeline/

Ready to go validation and governance controls for Azure Infrastructure as Code (IaC).

https://azure.github.io/PSRule.Rules.Azure/creating-your-pipeline/

**Comment by maxricketts** [comment]
I am having issues installing this with a custom self-hosted windows server 2019 image (error at the bottom).  

I had the same pipeline working with a Microsoft-hosted ubuntu image but need it to run on the self-hosted agent.

It seems to be installing the module.
Checking PSRule
[info] Using repository: PSGallery
[info] Installing PSRule: 2.6.0
Checking modules
> Checking module: PSRule.Rules.Azure
  - Installing module
  - Using version: 1.21.2
Checking environment
[info] Using PSRule: 2.6.0
[info] Using Extension: 2.6.0
[info] Using Task: 2.6.2211067
[info] Using Workspace: C:\a\1\s
[info] Using PWD: C:\a\_tasks\ps-rule-assert_8804fc31-b62f-4d49-b2a1-c80dc0879dae\2.6.2211067
[info] Using Path: C:\a\1\s
[info] Using Source: C:\a\1\s\.ps-rule\
[info] Using Baseline: 
[info] Using Conventions: 
[info] Using InputType: repository
[info] Using InputPath: C:\a\1\s\newImage\bicep\
[info] Using Option: 
[info] Using Outcome: 
[info] Using OutputFormat: Nunit3
[info] Using OutputPath: C:\a\1\s\reports\rule-report.xml

ERROR MESSAGE
##[error]Exit code 5 returned from process: file name 'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe', arguments '-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". ([scriptblock]::Create('if ([Console]::InputEncoding -is [Text.UTF8Encoding] -and [Console]::InputEncoding.GetPreamble().Length -ne 0) { [Console]::InputEncoding = New-Object Text.UTF8Encoding $false } if (!$PSHOME) { $null = Get-Item -LiteralPath ''variable:PSHOME'' } else { Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }')) 2>&1 | ForEach-Object { Write-Verbose $_.Exception.Message -Verbose } ; Import-Module -Name 'C:\a\_tasks\ps-rule-assert_8804fc31-b62f-4d49-b2a1-c80dc0879dae\2.6.2211067\ps_modules\VstsTaskSdk\VstsTaskSdk.psd1' -ArgumentList @{ NonInteractive = $true } -ErrorAction Stop ; $VerbosePreference = 'SilentlyContinue' ; $DebugPreference = 'SilentlyContinue' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''C:\a\_tasks\ps-rule-assert_8804fc31-b62f-4d49-b2a1-c80dc0879dae\2.6.2211067\powershell.ps1'''))"'.

**Reply by BernieWhite** [reply]
@maxricketts Is this related to question #1897?

**Reply by maxricketts** [reply]
@BernieWhite - I don't think so.  I have fixed it, but on a windows self-hosted agent it didn't like having to create a folder on outputPath.  if I remove reports/ it works all ok.

```yaml
 displayName: Analyze Azure template files
            inputs:
              modules: 'PSRule.Rules.Azure'
              inputType: "repository"
              inputPath: newImage/bicep/
              outputFormat: Nunit3
              outputPath: reports/rule-report.xml
              conventions: DebugTenant
```

**Reply by BernieWhite** [reply]
@maxricketts Ok sure. Yes, `reports/` should be automatically created. We currently have a bug with job summaries (microsoft/PSRule#1353) but this doesn't affect NUnit reports.

I've just double checked an it seems to be working for me so if the problem is persisting if you can log a bug with the error message that would be great.

**Reply by maxricketts** [reply]
@BernieWhite - I will as it has come back again after no changes, and a successful run.

Logged as #1901
